File: app/src/main/java/com/upclicks/ffcapp/repositories/AccountRepository.java
package com.upclicks.ffcapp.repositories;
import com.upclicks.ffcapp.data.ApiService;
import com.upclicks.ffcapp.models.Result;
import com.upclicks.ffcapp.models.requests.LoginRequest;
import com.upclicks.ffcapp.models.requests.SignUpRequest;
import com.upclicks.ffcapp.models.requests.UpdateProfileRequest;
import com.upclicks.ffcapp.models.response.City;
import com.upclicks.ffcapp.models.response.Profile;
import com.upclicks.ffcapp.session.UserSession;

import java.util.List;

import javax.inject.Inject;
import io.reactivex.rxjava3.core.Observable;
import okhttp3.MultipartBody;

public class AccountRepository {
    private ApiService apiService;

    @Inject
    public AccountRepository(ApiService apiService) {
        this.apiService = apiService;
    }

    //Login
    public Observable<Result<UserSession>> login(LoginRequest loginRequest){
        return apiService.login(loginRequest);
    }
    //SignUp
    public Observable<Result<UserSession>> signUp(SignUpRequest signUpRequest){
        return apiService.signUp(signUpRequest);
    }
    //Update Profile
    public Observable<Result<String>> updateProfile(UpdateProfileRequest updateProfileRequest){
        return apiService.updateProfile(updateProfileRequest);
    }
    //Get Cites
    public Observable<Result<List<City>>> getCites(){
        return apiService.getCites();
    }
    //Get Profile
    public Observable<Result<Profile>> getUserProfile(){
        return apiService.getUserProfile();
    }
    //Update Avatar
    public Observable<Result<String>> updateAvatar(MultipartBody.Part avatar){
        return apiService.updateAvatar(avatar);
    }
    //Change Password
    public Observable<Result<UserSession>> changePassword (Object o){
        return apiService.changePassword(o);
    }
}

File: app/src/main/java/com/upclicks/ffc/ui/authentication/repositories/AccountRepository.kt
package com.upclicks.ffc.ui.authentication.repositories

import com.upclicks.ffc.data.remote.ApiService
import com.upclicks.ffc.ui.authentication.model.response.*
import com.upclicks.ffc.ui.authentication.model.Session
import com.upclicks.ffc.ui.general.model.City
import com.upclicks.ffc.ui.general.model.Country
import io.reactivex.rxjava3.core.Observable
import kotlinx.coroutines.ExperimentalCoroutinesApi
import okhttp3.MultipartBody
import javax.inject.Inject
import com.upclicks.ffc.ui.authentication.model.request.*
import com.upclicks.ffc.ui.general.model.FeedbackRequest
import com.upclicks.ffc.data.remote.Result

@ExperimentalCoroutinesApi
class AccountRepository @Inject constructor(private val apiService: ApiService) {


    //Get Countries
    fun getCountries() : Observable<Result<List<Country>>> {
        return apiService.getCountries()
    }

    //Get Cities
    fun getCities() : Observable<Result<List<City>>> {
        return apiService.getCities()
    }


    //Create membership
    fun createMembership(createMemberShipRequest : CreateMemberShipRequest) : Observable<Result<MembershipResponse>> {
        return apiService.createMembership(createMemberShipRequest)
    }

    //Create membership
    fun validateResetPasswordCode(validateResetPasswordCodeRequest : ValidateResetPasswordCodeRequest) : Observable<Result<Boolean>> {
        return apiService.validateResetPasswordCode(validateResetPasswordCodeRequest)
    }

    //Verify membership
    fun verifyMembership(membershipIdentity : String, verificationCode : String) : Observable<Result<Boolean>> {
        return apiService.verifyMembership(membershipIdentity,verificationCode)
    }

    //Login
    fun login(loginRequest: LoginRequest): Observable<Result<Session>> {
        return apiService.authenticate(loginRequest)
    }

    //Verify Session
    fun verifySession(): Observable<Result<VerifySession>> {
        return apiService.verifySession()
    }

    //Complete profile
    fun completeProfile(completeProfileRequest: CompleteProfileRequest): Observable<Result<String>> {
        return apiService.completeProfile(completeProfileRequest)
    }

    //Update avatar
    fun updateAvatar(avatar: MultipartBody.Part): Observable<Result<String>> {
        return apiService.updateAvatar(avatar)
    }

    //Update profile
    fun updateProfile(updateProfileRequest: UpdateProfileRequest) : Observable<Result<String>> {
        return apiService.updateProfile(updateProfileRequest)
    }

    //Get my profile
    fun getMyProfile() : Observable<Result<VerifySession.Profile>> {
        return apiService.getMyProfile()
    }

    //Change Password
    fun changePassword(changePasswordRequest: ChangePasswordRequest) : Observable<Result<Boolean>> {
        return apiService.changePassword(changePasswordRequest)
    }


    //Forgot Password
    fun forgotPassword(emailAddress : String) : Observable<Result<ForgotPasswordResponse>> {
        return apiService.forgotPassword(emailAddress)
    }


    //Reset Password
    fun resetPassword(resetPasswordRequest: ResetPasswordRequest) : Observable<Result<Boolean>> {
        return apiService.resetPassword(resetPasswordRequest)
    }

    //Verify Account Identity
    fun verifyAccountIdentity(verifyIdRequest: VerifyIDRequest) : Observable<Result<String>> {
        return apiService.verifyAccountIdentity(verifyIdRequest)
    }

    //Get my profile
    fun getIdentityVerificationStatus(notifyId : String) : Observable<Result<IdentityVerificationStatus>> {
        return apiService.getIdentityVerificationStatus(notifyId)
    }

    //sendFeedback
    fun sendFeedback(feedbackRequest: FeedbackRequest) : Observable<Result<String>> {
        return apiService.sendFeedback(feedbackRequest)
    }

}